File: app/src/main/java/ir/hosseinabbasi/trademe/utils/CommonUtils.java
package ir.hosseinabbasi.trademe.utils;

import android.app.ProgressDialog;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;

import ir.hosseinabbasi.trademe.R;

/**
 * Created by Dr.jacky on 2018/03/09.
 */

public class CommonUtils {
    private static final String TAG = "CommonUtils";

    private CommonUtils() {
    }

    public static ProgressDialog showLoadingDialog(Context context) {
        ProgressDialog progressDialog = new ProgressDialog(context);
        progressDialog.show();
        if (progressDialog.getWindow() != null) {
            progressDialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        }
        progressDialog.setContentView(R.layout.progress_dialog);
        progressDialog.setIndeterminate(true);
        progressDialog.setCancelable(true);
        progressDialog.setCanceledOnTouchOutside(false);
        return progressDialog;
    }
}

File: app/src/main/kotlin/ir.hosseinabbasi.trademe/utils/CommonUtils.kt
package ir.hosseinabbasi.trademe.utils

import android.app.ProgressDialog
import android.content.Context
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import ir.hosseinabbasi.trademe.R

/**
 * Created by Dr.jacky on 2018/03/25.
 */
class CommonUtils constructor() {

    companion object {
        fun showLoadingDialog(context : Context) : ProgressDialog{
            val progressDialog = ProgressDialog(context)
            progressDialog.show()
            if (progressDialog.window != null) {
                progressDialog.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
            }
            progressDialog.setContentView(R.layout.progress_dialog)
            progressDialog.isIndeterminate = true
            progressDialog.setCancelable(true)
            progressDialog.setCanceledOnTouchOutside(false)
            return progressDialog
        }
    }
}